index.js

```diff
@@ -176,47 +186,34 @@ Client.ready().then(Connection => {
           
           return _Tx
         })
-        
-        // console.dir(Transactions[0], { depth: null })
-        // process.exit(1)
 
         bigquery.dataset(DATASET_NAME).table(TRANSACTION_TABLE_NAME).insert(Transactions)
           .then(r => {
             console.log(`Inserted rows`, r)
             LastLedger = Result.ledger_index
-            // process.exit(0)
           })
           .catch(err => {
             if (err && err.name === 'PartialFailureError') {
               if (err.errors && err.errors.length > 0) {
                 console.log('Insert errors:')
                 err.errors.forEach(err => console.dir(err, { depth: null }))
-                process.exit(1)
+                return safeHalt()
```

Comment:
FYI - I personally do not like `.then` and greatly prefer `async/await`. It took me forever to remember how to await a promise from within another promise, conditionally, without nesting. to me, the fact that `return`ing a promise from `then` will await the returned promise is too magical. `async/await` is much clearer

Reply:
have we considered using util.promisify to wrap this callback function `bigquery.dataset(DATASET_NAME).table(TRANSACTION_TABLE_NAME).insert` portion?